web/src/ui_util.ts

```diff
@@ -22,13 +23,47 @@ export function place_caret_at_end(el: HTMLElement): void {
     }
 }
 
-export function replace_emoji_with_text($element: JQuery): void {
+export function replace_emoji_name_with_emoji_unicode($element: JQuery): void {
     $element
         .find(".emoji")
         .text(function () {
+            // This is a custom emoji, we do not have corresponding emoji
+            // unicode for these so we return original markdown.
             if ($(this).is("img")) {
                 return $(this).attr("alt") ?? "";
             }
+
+            const emoji_name = $(this).text().slice(1, -1);
+            const emoji_code_hex_string = emoji.get_emoji_codepoint(emoji_name);
```

Comment:
Hmm, why are you doing this rather than parsing the CSS classes? The CSS classes are a much stronger guarantee of what codepoint we intend. 

There's a somewhat expectable situation where the name is stale (after an emoji name/codepoint mapping was changed) and thus logic won't work, but it should be very rare.